### PR TITLE
fix: fixed nested program parsing

### PR DESCRIPTION
**Pull request by grianbrcom**
Program end name check fixed.
Variable errors check fixed.

### CODE REVIEW

File: server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java [hunk 1]
```diff
@@ -14,13 +14,55 @@
...
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
 /**
  * This class represents program context in COBOL.
  */
 public class ProgramNode extends Node {
-  public ProgramNode(Location location) {
+  private VariableDefinitionDelegate variablesDelegate;
```
Comment (temanbrcom) [inline server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java:35]: Maybe, the name `variableDefinitionDelegate` will be more descriptive

**Review by temanbrcom — APPROVED** [review]

**Comment by temanbrcom** [comment]
The commit message body should start with a verb in the first form

**Comment by grianbrcom** [comment]
Will do.

File: server/src/test/java/org/eclipse/lsp/cobol/core/model/tree/NodeTest.java [hunk 2]
```diff
@@ -33,11 +33,11 @@
   @Test
   void getDepthFirstStream() {
     Node rootNode = new RootNode(LOCATION);
-    Node firstProg = new ProgramNode(LOCATION);
+    Node firstProg = new ProgramNode(LOCATION, null, null);
```
Comment (asatklichov) [inline server/src/test/java/org/eclipse/lsp/cobol/core/model/tree/NodeTest.java:36]: Can we put another constructor, so no need to add those "NULL"s.  

```
public ProgramEndNode(Locality locality) {
this(locality, null, null);
  }
```

Comment (grianbrcom) [reply server/src/test/java/org/eclipse/lsp/cobol/core/model/tree/NodeTest.java:36]: Yes, I'll make it with package visibility for tests only.

**Review by asatklichov — APPROVED** [review]
🏭

**Pull request merged**